### PR TITLE
Show a verification badge on the Room Member/User Profile screens.

### LINKED ISSUE #3413: [Task] Show in user's profile when they have been verified
Please see and discuss the details in the meta issue.

### PR DESCRIPTION
**Pull request by pixlwave**
This PR makes 2 changes to the Room Member and User Profile screens:

- Show a verified badge when verified.
- Show a disabled Verify button when the user isn't verified.

Requires https://github.com/matrix-org/matrix-rust-sdk/pull/4142, closes #3413

| Verified | Unverified |
| - | - |
| test_roomMemberDetailsScreen-iPhone-16-en-GB Verified-User | test_roomMemberDetailsScreen-iPhone-16-en-GB Other-User |

### CODE REVIEW

File: ElementX/Sources/Services/Client/ClientProxyProtocol.swift [hunk 1]
```diff
@@ -204,4 +204,6 @@ protocol ClientProxyProtocol: AnyObject, MediaLoaderProtocol {
     
     func pinUserIdentity(_ userID: String) async -> Result<Void, ClientProxyError>
     func resetIdentity() async -> Result<IdentityResetHandle?, ClientProxyError>
+    
+    func userIdentity(_ userID: String) async -> Result<UserIdentity?, ClientProxyError>
```
Comment (stefanceriu) [inline ElementX/Sources/Services/Client/ClientProxyProtocol.swift:208]: I feel like this is missing a verb 🙃

**Review by stefanceriu — APPROVED** [review]
lgtm ❇️

File: ElementX/Sources/Services/Client/ClientProxyProtocol.swift (hunk 1 above)
Comment (pixlwave) [reply ElementX/Sources/Services/Client/ClientProxyProtocol.swift:208]: Yeah it's weird isn't it. It is no different to `sessionVerificationController() -> SessionVerificationControllerProtocol` but it reads more weirdly. Maybe I'll make it `userIdentity(for: userID)`.

**Comment by pixlwave** [comment]
Ignoring enterprise tests as that same one is fine on normal. And SC is right these 2 screens are very similar, but thats fine for now.

**Pull request merged**